Plugins/Flow.Launcher.Plugin.Explorer/Search/SearchManager.cs

```diff
@@ -46,36 +47,39 @@ internal async Task<List<Result>> SearchAsync(Query query, CancellationToken tok
 
             var result = new HashSet<Result>(PathEqualityComparator.Instance);
 
-            if (ActionKeywordMatch(query, Settings.ActionKeyword.PathSearchActionKeyword) || ActionKeywordMatch(query, Settings.ActionKeyword.SearchActionKeyword))
+            if (ActionKeywordMatch(query, Settings.ActionKeyword.PathSearchActionKeyword) ||
+                ActionKeywordMatch(query, Settings.ActionKeyword.SearchActionKeyword))
             {
                 result.UnionWith(await PathSearchAsync(query, token).ConfigureAwait(false));
             }
 
-            if ((ActionKeywordMatch(query, Settings.ActionKeyword.IndexOnlySearchActionKeyword) || ActionKeywordMatch(query, Settings.ActionKeyword.SearchActionKeyword)) &&
-                 querySearch.Length > 0 &&
-                 !querySearch.IsLocationPathString())
+            if ((ActionKeywordMatch(query, Settings.ActionKeyword.IndexOnlySearchActionKeyword) ||
+                 ActionKeywordMatch(query, Settings.ActionKeyword.SearchActionKeyword)) &&
+                querySearch.Length > 0 &&
+                !querySearch.IsLocationPathString())
             {
-                result.UnionWith(await WindowsIndexFilesAndFoldersSearchAsync(query, querySearch, token).ConfigureAwait(false));
+                result.UnionWith(await WindowsIndexFilesAndFoldersSearchAsync(query, querySearch, token)
+                    .ConfigureAwait(false));
             }
 
             return result.ToList();
         }
 
         private bool ActionKeywordMatch(Query query, Settings.ActionKeyword allowedActionKeyword)
         {
-            if (query.ActionKeyword == settings.IndexOnlySearchActionKeyword)
-                return Settings.ActionKeyword.IndexOnlySearchActionKeyword == allowedActionKeyword && settings.EnabledIndexOnlySearchKeyword;
-
-            if (query.ActionKeyword == settings.PathSearchActionKeyword)
-                return Settings.ActionKeyword.PathSearchActionKeyword == allowedActionKeyword && settings.EnabledPathSearchKeyword;
+            var keyword = query.ActionKeyword.Length == 0 ? "*" : query.ActionKeyword;
 
-            if (query.ActionKeyword == settings.SearchActionKeyword)
-                return Settings.ActionKeyword.SearchActionKeyword == allowedActionKeyword;
-
-            
-            return (Settings.ActionKeyword.IndexOnlySearchActionKeyword == allowedActionKeyword && settings.EnabledIndexOnlySearchKeyword)
-                || (Settings.ActionKeyword.PathSearchActionKeyword == allowedActionKeyword && settings.EnabledPathSearchKeyword)
-                || settings.SearchActionKeyword == Query.GlobalPluginWildcardSign;
+            return allowedActionKeyword switch
+            {
+                Settings.ActionKeyword.SearchActionKeyword => settings.EnableSearchActionKeyword &&
```

Comment:
would this `Settings.ActionKeyword.SearchActionKeyword ` work if the action keyword has been updated several times, since you are not referencing the settings instance that contains the updated action keyword?

Reply:
The `Settings.ActionKeyword.SearchActionKeyword` is only an enum, so it is static

Reply:
oh yeah you are right, lol ignore me

Reply:
It is only used for matching which actionkeyword we want to check. `settings.SearchActionKeyword` is the actual reference we are checking.

Reply:
Yeah I actually did several rewrites before pushing this up so forgot which version I had settled on lol